packages/protocol/contracts/governance/EpochRewards.sol

```diff
@@ -0,0 +1,399 @@
+    setTargetValidatorEpochPayment(_targetValidatorEpochPayment);
+    targetVotingYieldParams.target = FixidityLib.wrap(targetVotingYieldInitial);
+    startTime = now;
+  }
+
+  /**
+   * @notice Returns the target voting yield parameters.
+   * @return The target, max, and adjustment factor for target voting yield.
+   */
+  function getTargetVotingYieldParameters() external view returns (uint256, uint256, uint256) {
+    TargetVotingYieldParameters storage params = targetVotingYieldParams;
+    return (params.target.unwrap(), params.max.unwrap(), params.adjustmentFactor.unwrap());
+  }
+
+  /**
+   * @notice Returns the rewards multiplier parameters.
+   * @return The max multiplier and under/over spend adjustment factors.
+   */
+  function getRewardsMultiplierParameters() external view returns (uint256, uint256, uint256) {
+    RewardsMultiplierParameters storage params = rewardsMultiplierParams;
+    return (
+      params.max.unwrap(),
+      params.adjustmentFactors.underspend.unwrap(),
+      params.adjustmentFactors.overspend.unwrap()
+    );
+  }
+
+  /**
+   * @notice Sets the target voting gold fraction.
+   * @param value The percentage of floating gold voting to target.
+   * @return True upon success.
+   */
+  function setTargetVotingGoldFraction(uint256 value) public onlyOwner returns (bool) {
+    require(value != targetVotingGoldFraction.unwrap() && value < FixidityLib.fixed1().unwrap());
+    targetVotingGoldFraction = FixidityLib.wrap(value);
+    emit TargetVotingGoldFractionSet(value);
+    return true;
+  }
+
+  /**
+   * @notice Returns the target voting gold fraction.
+   * @return The percentage of floating gold voting to target.
+   */
+  function getTargetVotingGoldFraction() external view returns (uint256) {
+    return targetVotingGoldFraction.unwrap();
+  }
+
+  /**
+   * @notice Sets the target per-epoch payment in Celo Dollars for validators.
+   * @param value The value in Celo Dollars.
+   * @return True upon success.
+   */
+  function setTargetValidatorEpochPayment(uint256 value) public onlyOwner returns (bool) {
+    require(value != targetValidatorEpochPayment);
+    targetValidatorEpochPayment = value;
+    emit TargetValidatorEpochPaymentSet(value);
+    return true;
+  }
+
+  /**
+   * @notice Sets the rewards multiplier parameters.
+   * @param max The max multiplier on target epoch rewards.
+   * @param underspendAdjustmentFactor Adjusts the multiplier on target epoch rewards when the
+   *   protocol is running behind the target gold supply.
+   * @param overspendAdjustmentFactor Adjusts the multiplier on target epoch rewards when the
+   *   protocol is running ahead of the target gold supply.
+   * @return True upon success.
+   */
+  function setRewardsMultiplierParameters(
+    uint256 max,
+    uint256 underspendAdjustmentFactor,
+    uint256 overspendAdjustmentFactor
+  )
+    public
+    onlyOwner
+    returns (bool)
+  {
+    require(
+      max != rewardsMultiplierParams.max.unwrap() ||
```

Comment:
Similar as above, consider first wrapping and using Fixidity comparisons.

Reply:
Same answer, if we wrap first (and overwrite storage) we can't do this check anymore